Task dev 9

NechayOlga opened the pull request:
This project was created to replace the sublines in two lines.

NechayOlga committed:
Add project of replacing of parts of lines.

NechayOlga committed:
Add solution for project called RandomReplacementPartsOfLines

NechayOlga committed:
Comments were added

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/Entrypoint.cs:
```diff
@@ -0,0 +1,29 @@
+{
+    class Entrypoint
+    {
+        static void Main(string[] args)
```
Comments

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/Outputer.cs:
```diff
@@ -0,0 +1,12 @@
+
+namespace RandomReplacementPartsOfLines
+{
+    class Outputer
```
Silly class, no need in it, it does no special things

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/SublinesReplacer.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace RandomReplacementPartsOfLines
+{
+    class SublinesReplacer
```
good idea but no need in a separate class

IgorKheidorov reviewed (changes requested):
Try to use simple char arrays

NechayOlga committed:
Add solution

NechayOlga committed:
Add changed solution

NechayOlga committed:
Add projectts to solution to resolve conflict

NechayOlga committed:
Class called SublinesReplacer was deleted

NechayOlga committed:
Class Outputer was deleted

NechayOlga committed:
Comments for class Entrypoint were added

NechayOlga committed:
Class for reading of text files was added

NechayOlga committed:
Name of method ReaderInitialLine in class LinesBuilder was changed to ReaderOfLines

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/RandomNumbersBuilder.cs:
```diff
@@ -0,0 +1,49 @@
+        // of the replaced subline.
+        // Parameter 1 in method Next() means the minimal
+        // possible value of the length.       
+        public int BuildLengthOfReplacedSubline(string initialLine, int beginningOfReplacedSubline)
```
Why do you implement two methods with similar functionality?
BuildLengthOfReplacedSubline
BuildLengthOfReplacingSubline

BuildBeginningOfReplacingSubline
BuildBeginningOfReplacedSubline

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/Entrypoint.cs:
```diff
@@ -0,0 +1,32 @@
+        {
+            //Creating initial and replacing lines
+            string line = @"e:\TAT\GIT-1\DEV-9\RandomReplacementPartsOfLines\Lines.txt";
+            LinesReader linesReader = new LinesReader();
```
there is no try- catch, initialLine and replacingLines can be null after file reading, exception is no not processed

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/LinesReader.cs:
```diff
@@ -0,0 +1,28 @@
+        {
+            using (StreamReader streamReader = new StreamReader(line))
+            {
+                initialLine = streamReader.ReadLine();
```
What will occur if initialLine is OK but replacingLine is null?

IgorKheidorov reviewed (changes requested):
Add comments to all classes and methods as for docs

NechayOlga committed:
Extra methods BuildLengthOfReplacedSubline
and BuildLengthOfReplacingSubline, BuildBeginningOfReplacingSubline, BuildBeginningOfReplacedSubline with the same functionality were modified to two methods called BuildReplacedSubline and BuildReplacingSubline

NechayOlga committed:
Exceptions for Empty file and null lines were added

NechayOlga committed:
Correct text file was added

NechayOlga committed:
Comments for all classes and methods were added

NechayOlga committed:
Extra line was deleted

NechayOlga committed:
Names of methods SublinesReplacing and ReaderOfLines were changed to ReplaceSublines and ReadOfLines

NechayOlga committed:
Comment to ReadOfLines was changed

NechayOlga committed:
The name of method ReadOfLines was changed to ReadLines

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/LinesBuilder.cs:
```diff
@@ -0,0 +1,57 @@
+        /// Method builds the replaced subline - subline, which will be replaced
+        /// </summary>
+        /// <returns>string replacedSubline</returns>
+        public string BuildReplacedSubline()
```
BuildReplacedSubline and  BuildReplacingSubline methods are the same, DRY

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/LinesReader.cs:
```diff
@@ -0,0 +1,37 @@
+        {
+            using (StreamReader streamReader = new StreamReader(line))
+            {
+                if(line == null)
```
"line" is a bad value  name for file

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/RandomNumbersBuilder.cs:
```diff
@@ -0,0 +1,39 @@
+        /// </summary>
+        /// <param name="line"> parametr, in which contains the value of current string</param>
+        /// <returns>int index of subline</returns>
+        public int BuildBeginningOfSubline(string line)
```
BuildLengthOfSubline and BuildBeginningOfSubline the same, DRY

IgorKheidorov commented on DEV-9/RandomReplacementPartsOfLines/RandomNumbersBuilder.cs:
```diff
@@ -0,0 +1,39 @@
+        /// </summary>
+        /// <param name="line"> parametr, in which contains the value of current string</param>
+        /// <returns>int length of subline</returns>       
+        public int BuildLengthOfSubline(string line, int beginningOfSubline)
```
Unnecessary method

IgorKheidorov reviewed (approved):
Not bad but too much repeating code, DRY

NechayOlga committed:
Unnecessary class RandomNumbersBuilder and methods with the same functionality were removed. The name of variable 'line', which contains the current value of line,  was changed to 'currentLine'.
